stdlib/mime/src/main/java/org/ballerinalang/mime/nativeimpl/GetBodyPartsAsChannel.java

```diff
@@ -36,16 +41,28 @@
 public class GetBodyPartsAsChannel extends BlockingNativeCallableUnit {
     @Override
     public void execute(Context context) {
-        BMap<String, BValue> byteChannelStruct;
-        BMap<String, BValue> entityStruct = (BMap<String, BValue>) context.getRefArgument(FIRST_PARAMETER_INDEX);
-        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
-        String multipartDataBoundary = MimeUtil.getNewMultipartDelimiter();
-        MultipartDataSource multipartDataSource = new MultipartDataSource(entityStruct, multipartDataBoundary);
-        multipartDataSource.serialize(outputStream);
-        EntityBodyChannel entityBodyChannel = new EntityBodyChannel(new ByteArrayInputStream(
-                outputStream.toByteArray()));
-        byteChannelStruct = BLangConnectorSPIUtil.createBStruct(context, IOConstants.IO_PACKAGE, BYTE_CHANNEL_STRUCT);
-        byteChannelStruct.addNativeData(IOConstants.BYTE_CHANNEL_NAME, new EntityWrapper(entityBodyChannel));
-        context.setReturnValues(byteChannelStruct);
+        try {
+            BMap<String, BValue> entityStruct = (BMap<String, BValue>) context.getRefArgument(FIRST_PARAMETER_INDEX);
+            String contentType = getContentTypeWithParameters(entityStruct);
+            if (isMultipart(contentType)) {
+                BString boundaryValue = HeaderUtil.extractBoundaryParameter(contentType);
+                String multipartDataBoundary = boundaryValue != null ? boundaryValue.toString() :
+                        getNewMultipartDelimiter();
+                ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+                MultipartDataSource multipartDataSource = new MultipartDataSource(entityStruct, multipartDataBoundary);
+                multipartDataSource.serialize(outputStream);
+                EntityBodyChannel entityBodyChannel = new EntityBodyChannel(new ByteArrayInputStream(
+                        outputStream.toByteArray()));
+                BMap<String, BValue> byteChannelStruct = BLangConnectorSPIUtil.createBStruct(context,
+                        IOConstants.IO_PACKAGE, BYTE_CHANNEL_STRUCT);
+                byteChannelStruct.addNativeData(IOConstants.BYTE_CHANNEL_NAME, new EntityWrapper(entityBodyChannel));
+                context.setReturnValues(byteChannelStruct);
+            } else {
+                context.setReturnValues(MimeUtil.createError(context, "Entity doesn't contain body parts"));
+            }
+        } catch (Throwable e) {
+            context.setReturnValues(MimeUtil.createError(context, "Error occurred while constructing a byte " +
```

Comment:
It's better if we can log the error here. Otherwise, we will lose the vital information about the error.